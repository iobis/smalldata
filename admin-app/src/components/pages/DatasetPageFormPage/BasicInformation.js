import Dropdown from '@smalldata/dwca-lib/src/components/form/Dropdown'
import InputText from '@smalldata/dwca-lib/src/components/form/InputText'
import PropTypes from 'prop-types'
import React from 'react'
import Textarea from '@smalldata/dwca-lib/src/components/form/Textarea'
import { useTranslation } from 'react-i18next'

export const licences = [
  'Creative Commons Attribution Non Commercial (CC-BY) 4.0 License',
  'Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License',
  'Public Domain (CC0 1.0)']
export const languages = [
  'English',
  'Dutch',
  'French',
  'Spanish']

export default function BasicInformation({ onChange, data }) {
  const { t } = useTranslation()
  const { title, licence, language, abstract } = data

  const updateField = (name, value) => {
    const newSelection = { ...data, [name]: value }
    onChange(newSelection)
  }

  return (
    <div className="basic-information">
      <InputText
        className="title"
        name="datasetPageFormPage.basicInformation.title"
        onChange={value => updateField('title', value)}
        value={title}/>
      <div className="column field licence">
        <label className="label">
          {t('datasetPageFormPage.basicInformation.licence.label')}
        </label>
        <Dropdown
          onChange={value => updateField('licence', value)}
          options={licences}
          value={licence}/>
      </div>
      <div className="column field language">
        <label className="label">
          {t('datasetPageFormPage.basicInformation.language.label')}
        </label>
        <Dropdown
          onChange={value => updateField('language', value)}
          options={languages}
          value={language}/>
      </div>
      <Textarea
        className="abstract is-9"
        name="datasetPageFormPage.basicInformation.abstract"
        onChange={value => updateField('abstract', value)}
        value={abstract}/>
    </div>
  )
}

export const basicInformationShape = {
  title:                  PropTypes.string.isRequired,
  licence:                PropTypes.oneOf(licences).isRequired,
  language:               PropTypes.oneOf(languages).isRequired,
  abstract:               PropTypes.string.isRequired
}

BasicInformation.propTypes = {
  data:     PropTypes.shape(basicInformationShape).isRequired,
  onChange: PropTypes.func.isRequired
}
