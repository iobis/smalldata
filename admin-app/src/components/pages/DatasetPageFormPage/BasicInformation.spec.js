import BasicInformation, { languages, licences } from './BasicInformation'
import React from 'react'
import { mount } from 'enzyme'

describe('BasicInformation', () => {
  it('renders correctly', () => {
    expect(mount(
      <BasicInformation
        data={getDefaultFilledData()}
        onChange={jest.fn()}/>))
      .toMatchSnapshot()
  })

  it('returns updated title when when changing title', () => {
    const onChange = jest.fn()
    const wrapper = mount(
      <BasicInformation
        data={getDefaultFilledData()}
        onChange={onChange}/>)

    wrapper.find('.basic-information .title input').simulate('change', { target: { value: 'new-title' } })
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenNthCalledWith(1, { ...getDefaultFilledData(), title: 'new-title' })
  })
})

function getDefaultFilledData() {
  return {
    title:                  'title',
    licence:                licences[0],
    language:               languages[0],
    abstract:               'abstract'
  }
}
