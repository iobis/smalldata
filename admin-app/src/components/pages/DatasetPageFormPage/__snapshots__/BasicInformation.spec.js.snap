// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInformation renders correctly 1`] = `
<BasicInformation
  data={
    Object {
      "abstract": "abstract",
      "language": "English",
      "licence": "Creative Commons Attribution Non Commercial (CC-BY) 4.0 License",
      "title": "title",
    }
  }
  onChange={[MockFunction]}
>
  <div
    className="basic-information"
  >
    <InputText
      className="title"
      name="datasetPageFormPage.basicInformation.title"
      onChange={[Function]}
      value="title"
    >
      <div
        className="column field title"
      >
        <label
          className="label"
        >
          datasetPageFormPage.basicInformation.title.label
        </label>
        <input
          className="input"
          onChange={[Function]}
          type="text"
          value="title"
        />
      </div>
    </InputText>
    <div
      className="column field licence"
    >
      <label
        className="label"
      >
        datasetPageFormPage.basicInformation.licence.label
      </label>
      <Dropdown
        onChange={[Function]}
        options={
          Array [
            "Creative Commons Attribution Non Commercial (CC-BY) 4.0 License",
            "Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License",
            "Public Domain (CC0 1.0)",
          ]
        }
        value="Creative Commons Attribution Non Commercial (CC-BY) 4.0 License"
      >
        <div
          className="dropdown"
          onClick={[Function]}
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button"
            >
              <span
                className="selected-value"
              >
                Creative Commons Attribution Non Commercial (CC-BY) 4.0 License
              </span>
              <span
                className="icon is-small"
              >
                <FontAwesomeIcon
                  className="angle-down"
                  icon="angle-down"
                >
                  <i
                    className="angle-down"
                    icon="angle-down"
                  />
                </FontAwesomeIcon>
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item is-active"
                key="Creative Commons Attribution Non Commercial (CC-BY) 4.0 License"
                onClick={[Function]}
              >
                Creative Commons Attribution Non Commercial (CC-BY) 4.0 License
              </div>
              <div
                className="dropdown-item"
                key="Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License"
                onClick={[Function]}
              >
                Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License
              </div>
              <div
                className="dropdown-item"
                key="Public Domain (CC0 1.0)"
                onClick={[Function]}
              >
                Public Domain (CC0 1.0)
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </div>
    <div
      className="column field language"
    >
      <label
        className="label"
      >
        datasetPageFormPage.basicInformation.language.label
      </label>
      <Dropdown
        onChange={[Function]}
        options={
          Array [
            "English",
            "Dutch",
            "French",
            "Spanish",
          ]
        }
        value="English"
      >
        <div
          className="dropdown"
          onClick={[Function]}
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button"
            >
              <span
                className="selected-value"
              >
                English
              </span>
              <span
                className="icon is-small"
              >
                <FontAwesomeIcon
                  className="angle-down"
                  icon="angle-down"
                >
                  <i
                    className="angle-down"
                    icon="angle-down"
                  />
                </FontAwesomeIcon>
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item is-active"
                key="English"
                onClick={[Function]}
              >
                English
              </div>
              <div
                className="dropdown-item"
                key="Dutch"
                onClick={[Function]}
              >
                Dutch
              </div>
              <div
                className="dropdown-item"
                key="French"
                onClick={[Function]}
              >
                French
              </div>
              <div
                className="dropdown-item"
                key="Spanish"
                onClick={[Function]}
              >
                Spanish
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </div>
    <Textarea
      className="abstract is-9"
      name="datasetPageFormPage.basicInformation.abstract"
      onChange={[Function]}
      value="abstract"
    >
      <div
        className="column field abstract is-9"
      >
        <label
          className="label"
        >
          datasetPageFormPage.basicInformation.abstract.label
        </label>
        <textarea
          className="textarea"
          onChange={[Function]}
          rows={5}
          value="abstract"
        />
      </div>
    </Textarea>
  </div>
</BasicInformation>
`;
